### PR TITLE
Support multiple Metazoa gene trees

### PR DESCRIPTION
**Pull request by azangru**
Metazoa team has split their gene trees into three clusters: 'default', 'protostomes', and 'insects' (the insects are currently unavailable). This PR attempts to show links to pages with two different trees, depending on whether a species has them.

### Examples
- Honeybee (has both default and protostomes trees): http://wp-np2-1e.ebi.ac.uk:8440/Apis_mellifera/Gene/Protostomes_Tree?g=LOC726692;r=CM009944.2:6529304-6531367
- Actinia equina (has only the default tree): http://wp-np2-1e.ebi.ac.uk:8440/Actinia_equina_GCA_011057435.1/Gene/Compara_Tree?g=EGACTEQ4350041359;r=WHPX01001324.1:62642-66571;t=EGACTEQ4350041359-RA
- Trichuris muris (has only the protostomes tree): http://wp-np2-1e.ebi.ac.uk:8440/Trichuris_muris_prjeb126/Gene/Protostomes_Tree?g=WBGene00286032;r=TMUE_LG2:6899491-6904247;t=TMUE_2000008757.1

Note that there is still room for improvement:

- this notification contains a very confusing message:

image

- the Strain section of the left-hand navigation.

### CODE REVIEW

File: modules/EnsEMBL/Web/Configuration/Gene.pm [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+sub has_default_gene_tree {
+  my ($self, $clusterset_ids) = @_;
+
+  return any { $_ eq "default" } @$clusterset_ids;
+}
+
+sub has_protostomes_gene_tree {
+  my ($self, $clusterset_ids) = @_;
+
+  warn Dumper("Clusterset ids in subroutine", $clusterset_ids);
```
Comment (jyothishnt) [inline modules/EnsEMBL/Web/Configuration/Gene.pm:63]: remove warn?

File: modules/EnsEMBL/Web/Configuration/Gene.pm [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=cut
+
+package EnsEMBL::Web::Configuration::Gene;
+
+use List::MoreUtils qw(any);
+
+use Data::Dumper;
```
Comment (jyothishnt) [inline modules/EnsEMBL/Web/Configuration/Gene.pm:24]: remove?

File: modules/EnsEMBL/Web/Configuration/Gene.pm [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  $genetree_menu->set('availability', $self->has_default_gene_tree($clusterset_ids));
+
+  $genetree_menu->after($self->create_node('Protostomes_Tree', 'Gene tree (Protostomes)',
+    [qw( image EnsEMBL::Web::Component::Gene::ProtostomesTree )],
+    { 'availability' => $self->has_protostomes_gene_tree($clusterset_ids) }
+  ));
+
+}
+
+sub has_default_gene_tree {
```
Comment (jyothishnt) [inline modules/EnsEMBL/Web/Configuration/Gene.pm:54]: Should we move these subroutines into Object/Gene.pm and then at line 49, just say `has_protostomes_gene_gree` like in other places?

**Comment by jyothishnt** [comment]
Can we also get the "Strain" node and remove it from the left hand menu if it is not needed?

File: modules/EnsEMBL/Web/Configuration/Gene.pm (hunk 3 above)
Comment (azangru) [reply modules/EnsEMBL/Web/Configuration/Gene.pm:54]: I don't see what benefit this change would bring. We would be spreading this clusterset logic across multiple modules, when so far there's only one module that is concerned about it.

File: modules/EnsEMBL/Web/Configuration/Gene.pm (hunk 1 above)
Comment (azangru) [reply modules/EnsEMBL/Web/Configuration/Gene.pm]: ✅

File: modules/EnsEMBL/Web/Configuration/Gene.pm (hunk 2 above)
Comment (azangru) [reply modules/EnsEMBL/Web/Configuration/Gene.pm]: ✅

**Review by jyothishnt — APPROVED** [review]

File: modules/EnsEMBL/Web/Configuration/Gene.pm [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+sub modify_tree {
+  my $self         = shift;
+  my $hub          = $self->hub;
+  my $species_defs = $hub->species_defs;
+  my $species = $hub->species;
+
+  my $species_production_name = $species_defs->SPECIES_PRODUCTION_NAME;
+
+  $Data::Dumper::Maxdepth = 6;
```
Comment (veidenberg) [inline modules/EnsEMBL/Web/Configuration/Gene.pm:32]: This might cause compilation errors.

Comment (veidenberg) [inline modules/EnsEMBL/Web/Configuration/Gene.pm:32]: This might cause compilation errors.

Comment (azangru) [reply modules/EnsEMBL/Web/Configuration/Gene.pm:32]: Dang it! 😠

Comment (azangru) [reply modules/EnsEMBL/Web/Configuration/Gene.pm:32]: Deleted and pushed (https://github.com/EnsemblGenomes/eg-web-metazoa/commit/6b5c499751b1125c348f892d157e9df8301ba5e7)

**Pull request merged**